### PR TITLE
Extract lib track selection

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: *(direct link to the issue)*
Includes all commits of #4211 which awaits QA approval and merge

This is another step in unblocking the extraction of a toolkit neutral
library for destructive effect application and the registry of effects.

A library for changes in binary selection states of tracks in a project,
and for project sync-lock state and the sync-lock grouping logic.

TrackFocus would make sense as a part of this too, but more untangling
work must be done first before that can be toolkit-neutral, not dependent
on wxWidgets' interface for accessibility.

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

**Comment by Paul-Licameli** [comment]
Rebased without changes onto more recent master.

**Review by crsib — APPROVED** [review]

**Comment by dozzzzer** [comment]
Tested on macOS Ventura and Windows 11

**Comment by Paul-Licameli** [comment]
Rebased and resolved minor conflicts before merging, but I'm waiting for the present CI problems to be cleared.  The previous merge into master has not built successfully yet.

**Pull request merged**
